x-pack/plugins/actions/server/usage/actions_telemetry.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { APICaller } from 'kibana/server';
+
+export async function getTotalCount(callCluster: APICaller, kibanaIndex: string) {
+  const scriptedMetric = {
+    scripted_metric: {
+      init_script: 'state.types = [:]',
+      map_script: `
+        String actionType = doc['action.actionTypeId'].value;
+        state.types.put(actionType, state.types.containsKey(actionType) ? state.types.get(actionType) + 1 : 1);
+      `,
+      // Combine script is executed per cluster, but we already have a key-value pair per cluster.
+      // Despite docs that say this is optional, this script can't be blank.
+      combine_script: 'return state',
+      // Reduce script is executed across all clusters, so we need to add up all the total from each cluster
+      // This also needs to account for having no data
+      reduce_script: `
+        Map result = [:];
+        for (Map m : states.toArray()) {
+          if (m !== null) {
+            for (String k : m.keySet()) {
+              result.put(k, result.containsKey(k) ? result.get(k) + m.get(k) : m.get(k));
+            }
+          }
+        }
+        return result;
+      `,
+    },
+  };
+
+  const searchResult = await callCluster('search', {
+    index: kibanaIndex,
+    rest_total_hits_as_int: true,
+    body: {
+      query: {
+        bool: {
+          filter: [{ term: { type: 'action' } }],
+        },
+      },
+      aggs: {
+        byActionTypeId: scriptedMetric,
+      },
+    },
+  });
+
+  return {
+    countTotal: Object.keys(searchResult.aggregations.byActionTypeId.value.types).reduce(
+      (total: number, key: string) =>
+        parseInt(searchResult.aggregations.byActionTypeId.value.types[key], 0) + total,
+      0
+    ),
+    countByType: searchResult.aggregations.byActionTypeId.value.types,
+  };
+}
+
+export async function getInUseTotalCount(callCluster: APICaller, kibanaIndex: string) {
```

Comment:
It seems like we could probably do all these aggs in a single search, but I doubt there's a great reason to do that, since they won't be running all that often.  And it no doubt easier to deal with a small set of semi-complex searchs, rather than run a single search with a bunch of semi-complex aggs in it.

Reply:
I had the same thoughts, when was writing this code, but decided to keep it separately for now.